src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        str += this[i];
+      } else if (this[i] === null || this[i] === undefined) {
+        str += ',';
```

Comment:
I think you should use separator here
```suggestion
        str += separator;
```

Reply:
Yep

Reply:
Sure, I forgot to change that...